pkg/vsphere/disk/disk.go

```diff
@@ -45,6 +45,40 @@ type Filesystem interface {
 	Unmount(path string) error
 }
 
+type ReferenceCount struct {
+	owner      string
+	refname    string
+	references int
+}
+
+func NewReferenceCount(o, n string) *ReferenceCount {
+	return &ReferenceCount{
+		owner:      o,
+		refname:    n,
+		references: 0,
+	}
+}
+
+func (r *ReferenceCount) Increment() {
```

Comment:
I would consider using atomics in this case you would eliminate a whole class of a potential concurrency problems.